dataflux_pytorch/multipart_upload/multipart.py

```diff
@@ -0,0 +1,300 @@
+        and size of the blob. The remote API has restrictions on the minimum
+        and maximum size allowable, see: `https://cloud.google.com/storage/quotas#requests`
+
+    :type deadline: int
+    :param deadline:
+        The number of seconds to wait for all threads to resolve. If the
+        deadline is reached, all threads will be terminated regardless of their
+        progress and `concurrent.futures.TimeoutError` will be raised. This can
+        be left as the default of `None` (no deadline) for most use cases.
+
+    :type max_workers: int
+    :param max_workers:
+        The maximum number of workers to create to handle the workload.
+
+        How many workers is optimal depends heavily on the specific use case,
+        and the default is a conservative number that should work okay in most
+        cases without consuming excessive resources.
+
+    :type checksum: str
+    :param checksum:
+        (Optional) The checksum scheme to use: either "md5", "crc32c" or None.
+        Each individual part is checksummed. At present, the selected checksum
+        rule is only applied to parts and a separate checksum of the entire
+        resulting blob is not computed. Please compute and compare the checksum
+        of the file to the resulting blob separately if needed, using the
+        "crc32c" algorithm as per the XML MPU documentation.
+
+    :type timeout: float or tuple
+    :param timeout:
+        (Optional) The amount of time, in seconds, to wait
+        for the server response.  See: :ref:`configuring_timeouts`
+
+    :type retry: google.api_core.retry.Retry
+    :param retry: (Optional) How to retry the RPC. A None value will disable
+        retries. A `google.api_core.retry.Retry` value will enable retries,
+        and the object will configure backoff and timeout options. Custom
+        predicates (customizable error codes) are not supported for media
+        operations such as this one.
+
+        This function does not accept `ConditionalRetryPolicy` values because
+        preconditions are not supported by the underlying API call.
+
+        See the retry.py source code and docstrings in this package
+        (`google.cloud.storage.retry`) for information on retry types and how
+        to configure them.
+
+    :raises: :exc:`concurrent.futures.TimeoutError` if deadline is exceeded.
+    """
+
+    bucket = blob.bucket
+    client = blob.client
+    transport = blob._get_transport(client)
+
+    hostname = _get_host_name(client._connection)
+    url = "{hostname}/{bucket}/{blob}".format(hostname=hostname,
+                                              bucket=bucket.name,
+                                              blob=_quote(blob.name))
+
+    base_headers, object_metadata, content_type = blob._get_upload_arguments(
+        client, content_type, filename=None, command="tm.upload_sharded")
+    headers = {**base_headers, **_headers_from_metadata(object_metadata)}
```

Comment:
Do we need to make sure the "dataflux" user-agent header is set here, or is it already included in the base_headers?